tests/components/sensor/test_pilight.py

```diff
@@ -0,0 +1,120 @@
+"""The tests for the Pilight sensor platform."""
+import logging
+
+from homeassistant.bootstrap import setup_component
+import homeassistant.components.sensor as sensor
+from homeassistant.components import pilight
+
+from tests.common import get_test_home_assistant, assert_setup_component
+
+HASS = None
+
+
+def fire_pilight_message(protocol, data):
+    """Fire the fake pilight message."""
+    message = {pilight.ATTR_PROTOCOL: protocol}
+    message.update(data)
+    HASS.bus.fire(pilight.EVENT, message)
+
+
+def setUpModule():   # pylint: disable=invalid-name
+    """Initialize a Home Assistant server."""
+    global HASS
+
+    HASS = get_test_home_assistant()
+    HASS.config.components = ['pilight']
+
+
+def tearDownModule():   # pylint: disable=invalid-name
+    """Stop the Home Assistant server."""
+    HASS.stop()
+
+
+def test_sensor_value_from_code():
+    """Test the setting of value via pilight."""
+    with assert_setup_component(1):
```

Comment:
Thank you for your input, but I guess I have to phrase the problem better ;-)
- I have 3 test cases in `tests/components/sensor/test_pilight.py`
- Each test case runs successfully if none of the other test cases setup components
- On the second call to `setup component` on the same HASS instance the components does not seem to be setup. The call `HASS.states.get('sensor.test_2')` returns `None` as you can see in the [travis log](https://travis-ci.org/home-assistant/home-assistant/jobs/168341916#L1612).